cmake/version_app.cmake

```diff
@@ -0,0 +1,40 @@
+if(${APP_VERSION_NUMBER})
```

Comment:
why do we need a dedicated `version_app.cmake`.

Also, if we do need it, then it should have copyright and license header.

Reply:
Because the user application might not reside in a git repo, however zephyr and ncs do

Reply:
> Because the user application might not reside in a git repo, however zephyr and ncs do

I don't see why the question of application being in a git repo or not should decide whether the version files cannot be unified ?

If this is because you think the `git` handling doesn't belong in `version.cmake` then I prefer that upstream is updated to support a common `git` CMake function, which can then be reused downstream, and only the need for a single `version.cmake` file.

Reply:
It is need to know if a commit file is generated and what message to show